webapi/src/main/scala/org/knora/webapi/routing/admin/AuthenticatorServiceLive.scala

```diff
@@ -27,9 +27,10 @@ case class AuthenticatorServiceLive(actorDeps: ActorDeps, appConfig: AppConfig,
   private implicit val ec: ExecutionContext = actorDeps.executionContext
 
   private val authCookieName = Authenticator.calculateCookieName(appConfig)
+
   override def getUser(request: Request): Task[UserADM] =
     ZIO
-      .succeed(extractCredentialsFromRequest(request, authCookieName))
+      .attempt(extractCredentialsFromRequest(request, authCookieName))
```

Comment:
The method signature `def extractCredentialsFromRequest(Request, String)(implicit StringFormatter): Option[KnoraCredentialsV2]` lets me would assume it does not throw an `Throwable`. Your change is not necessary I would assume. 

Nevertheless, if you have experienced an exception here we should rather look into fixing the `extractCredentialsFromRequest`, shouldn't we?

Reply:
Sadly, it does throw (quite a lot, actually...).  
I generally agree with you, but the Authenticator will be simplified so much in the course of the authentication initiative, that I thought it doesn't make sense to touch this now (and `zio.attempt()` seemed like an easy fix for now).

Reply:
How about returning a `Try` from the `extractCredentialsFromRequest` by inside it simply wrapping the code into one and then use `ZIO.fromTry?`

Reply:
I would still like to understand why it would throw. The `org.knora.webapi.routing.admin.AuthenticatorServiceLiveSpec` does not account for any of these cases.

Reply:
it's easy enough to write a `AuthenticatorServiceLiveSpec ` test that blows up with an exception (not that I think it's well designed that this is the case). Those cases just haven't been covered by the tests.  
But wouldn't it make more sense to return  a `Task[Option[KnoraCredentialsV2]]` in `extractCredentialsFromRequest `? In the ZIO world we shouldn't need Try, I think

Reply:
The method `extractCredentialsFromRequest` does not return a ZIO, the return type of `getUser` would stay the same.